include/utility.h

```diff
@@ -249,19 +263,19 @@ class ParamServer
         sensor_msgs::Imu imu_out = imu_in;
         // rotate acceleration
         Eigen::Vector3d acc(imu_in.linear_acceleration.x, imu_in.linear_acceleration.y, imu_in.linear_acceleration.z);
-        acc = extRot * acc;
+        acc = extRot * imuAcc.inverse() * acc;
         imu_out.linear_acceleration.x = acc.x();
         imu_out.linear_acceleration.y = acc.y();
         imu_out.linear_acceleration.z = acc.z();
         // rotate gyroscope
         Eigen::Vector3d gyr(imu_in.angular_velocity.x, imu_in.angular_velocity.y, imu_in.angular_velocity.z);
-        gyr = extRot * gyr;
+        gyr = extRot * imuGyr.inverse() * gyr;
         imu_out.angular_velocity.x = gyr.x();
         imu_out.angular_velocity.y = gyr.y();
         imu_out.angular_velocity.z = gyr.z();
         // rotate roll pitch yaw
         Eigen::Quaterniond q_from(imu_in.orientation.w, imu_in.orientation.x, imu_in.orientation.y, imu_in.orientation.z);
-        Eigen::Quaterniond q_final = q_from * extQRPY;
+        Eigen::Quaterniond q_final = q_from * imuQMag.inverse() * extQRPY;
```

Comment:
Nitpicky, but I would call the param `imuOrient` as the orientation output is generally the result of fusion of mag, accelerometer and gyro data, not just the magnetmeter.

Besides that, I think the existing `extRPY` param is a bit questionable as it seems to unnecessarily compose NED->ENU conversion and `extRot`. I would suggest something like this instead
```C++
Eigen::Quaterniond imuworld_Q_imu(imu_in.orientation.w, imu_in.orientation.x,
                                  imu_in.orientation.y, imu_in.orientation.z);
Eigen::Quaterniond enu_Q_lidar = enu_Q_imuworld * imuworld_Q_imu * imuQOrient.inverse() * imu_Q_lidar;
imu_out.orientation.x          = enu_Q_lidar.x();
imu_out.orientation.y          = enu_Q_lidar.y();
imu_out.orientation.z          = enu_Q_lidar.z();
imu_out.orientation.w          = enu_Q_lidar.w();
```
Where `imu_Q_lidar = extRot.transpose()`, 
```python
enu_R_imuworld = [0,  1,  0,
                  1,  0,  0,
                  0,  0, -1]
```
and `imuQOrient` is just the renamed `imuQMag`.

Reply:
Actually, the transform math I suggested does not appear to work correctly, unfortunately. Not sure yet why. This one closer to the existing expression does though, even for significant rotation + translation:
```C++
Eigen::Quaterniond world_Q_imu(imu_in.orientation.w, imu_in.orientation.x,
                                  imu_in.orientation.y, imu_in.orientation.z);
Eigen::Quaterniond enu_Q_lidar = world_Q_imu * imuQOrient.inverse() * imu_Q_lidar;
```
where now instead of `enu_Q_imuworld`
```python
imuOrient = [0,  1,  0,
             1,  0,  0,
             0,  0, -1]
```
for the Microstrain IMU.

Reply:
The reason why @TixiaoShan has multiple extrinsic rotation matrices (`extRPY` and `extRot`) is because his AHRS does not provide IMU and RPY data in the same frame, so he must rotate them independently

Reply:
And, as I mentioned in [this comment](https://github.com/TixiaoShan/LIO-SAM/pull/173#discussion_r549131026), the matrix you are naming `imuOrient` is not orthonormal

Reply:
Ok, thanks for clarifying that. If the orientation correction matrix is not strictly orthonormal, then the calculation will need to be done on full matrices instead of quaternions, though, no?

Reply:
Ah yes, good spot! I'll amend this